test/e2e/tests/test_replicationgroup.py

```diff
@@ -244,7 +261,7 @@ def test_rg_delete(self, rg_cmd_update_input, rg_deletion_waiter):
 
         resource = k8s.get_resource(reference)
         assert resource['metadata']['deletionTimestamp'] is not None
-        # uncomment when reconciler->cleanup() invokes patchResource()
+        # TODO: uncomment when reconciler->cleanup() invokes patchResource()
```

Comment:
reconciler->cleanup() does not invoke `patchResource()` because it finalizes the deletion of the resource from Kubernetes.

Reply:
That discussion is out of scope of this PR, I just added a `TODO` here since it felt like the comment needed one.

More details:
https://github.com/aws-controllers-k8s/runtime/pull/21
https://github.com/aws-controllers-k8s/elasticache-controller/pull/34

@kumargauravsharma can elaborate if needed.